### ISSUE TITLE
Galacticraft 3.0.12.454 bug with arrows

### ISSUE THREAD

**Issue by Somriure** [micdoodle8/Galacticraft#2189]
Hi. I dont use mods with animation and e.t.c. Render API and Player API dont fix that.
Arrows fly around player.
![2016-05-04_00 32 58](https://cloud.githubusercontent.com/assets/18958092/14999276/e0634d9e-1190-11e6-937a-584d3a0ba8dd.png)

**Comment by EzerArch** [comment]
Yes, it's something related to the arrow physics, and happens when GC is installed.
![2016-05-03_22 24 59](https://cloud.githubusercontent.com/assets/6743303/15003043/f3af80ba-117d-11e6-9f5b-caa4dccdff88.png)

**Comment by radfast** [comment]
Hi guys, are we saying this is a hitbox issue?

I don't understand the screenshots: at the time the screenshot is taken, are those arrows moving or does the game think they hit the player?

**Comment by EzerArch** [comment]
> Hi guys, are we saying this is a hitbox issue?

Not exactly. It's the stuck arrows. Some of them are rendered at incorrect positions, somewhat 2 blocks up.

To reproduce:
1. Place a dispenser with arrows (half stack will suffice, or a skeleton)
2. Stand right in front of it
3. Get shot
4. After you have been shot for the 10th or 11th time, some stuck arrows will move 2 blocks up.
5. Remove GC. All arrows will be rendered correctly.

**Comment by radfast** [comment]
Which Forge version?  I'm not seeing this in my tests, all arrows are in the correct position.

**Comment by EzerArch** [comment]
> Which Forge version?

1614.

> I'm not seeing this in my tests, all arrows are in the correct position.

You need to be shot a lot. The 1st 10-11 stuck arrows will be ok, but the next ones will move 2 blocks up.

![2017-03-19_15 37 33](https://cloud.githubusercontent.com/assets/6743303/24083673/51b3dda4-0cba-11e7-99f9-15baf26f721f.png)

Let me test with GC+NEI only, on forge 1614 (I'm using this version b/c LexManos fixed a few issues with dispensers).

**Comment by EzerArch** [comment]
Double-posting with relevant information.

Forge 1614
CodeChickenCore-1.7.10-1.0.7.47-universal.jar
NotEnoughItems-1.7.10-1.0.5.120-universal.jar
GalacticraftCore-1.7-3.0.12.481.jar
Galacticraft-Planets-1.7-3.0.12.481.jar
MicdoodleCore-1.7-3.0.12.481.jar

**Video (2:00, sped up 1.5x):**  https://youtu.be/ZZbZkSgISF4 (uploading, ready in 5 min)

0:00-1:00 with GC.
1:00-2:00 without GC.

**Comment by radfast** [comment]
Likely caused by this code in vanilla's `LayerArrow`:

>              ModelRenderer modelrenderer = this.renderer.getMainModel().getRandomModelBox(random);
                ModelBox modelbox = (ModelBox)modelrenderer.cubeList.get(random.nextInt(modelrenderer.cubeList.size()));

**Comment by EzerArch** [comment]
I've just done a test on 1.12.2.

The arrows start to stick high up after the 8~10th arrow stuck in the player model.

![2020-02-17_20 20 53](https://user-images.githubusercontent.com/6743303/74691893-7c0e3580-51c3-11ea-943a-62b6477ac99a.png)
![2020-02-17_20 21 47](https://user-images.githubusercontent.com/6743303/74691898-7e708f80-51c3-11ea-8c1c-7aca2f6e0bee.png)
![2020-02-17_20 21 59](https://user-images.githubusercontent.com/6743303/74691899-7fa1bc80-51c3-11ea-871e-65e744e620f9.png)

Forge 2847
GalacticraftCore-1.12.2-4.0.2.244
Galacticraft-Planets-1.12.2-4.0.2.244
MicdoodleCore-1.12.2-4.0.2.244
No resource packs
